pydis_site/apps/api/viewsets/bot/user.py

```diff
@@ -118,4 +168,69 @@ class UserViewSet(BulkCreateModelMixin, ModelViewSet):
     """
 
     serializer_class = UserSerializer
-    queryset = User.objects
+    queryset = User.objects.all()
+    pagination_class = UserListPagination
+
+    def get_serializer(self, *args, **kwargs) -> ModelSerializer:
+        """Set Serializer many attribute to True if request body contains a list."""
+        if isinstance(kwargs.get('data', {}), list):
+            kwargs['many'] = True
+
+        return super().get_serializer(*args, **kwargs)
+
+    @action(detail=False, methods=["PATCH"], name='user-bulk-patch')
+    def bulk_patch(self, request: Request) -> Response:
+        """
+        Update multiple User objects in a single request.
+
+        ## Route
+        ### PATCH /bot/users/bulk_patch
+        Update all users with the IDs.
+        `id` field is mandatory, rest are optional.
+
+        #### Request body
+        >>> [
+        ...     {
+        ...     'id': int,
+        ...     'name': str,
+        ...     'discriminator': int,
+        ...     'roles': List[int],
+        ...     'in_guild': bool
+        ...     },
+        ...     {
+        ...     'id': int,
+        ...     'name': str,
+        ...     'discriminator': int,
+        ...     'roles': List[int],
+        ...     'in_guild': bool
```

Comment:
The dict items should be indented.
```suggestion
        ...         'id': int,
        ...         'name': str,
        ...         'discriminator': int,
        ...         'roles': List[int],
        ...         'in_guild': bool
        ...     },
        ...     {
        ...         'id': int,
        ...         'name': str,
        ...         'discriminator': int,
        ...         'roles': List[int],
        ...         'in_guild': bool
```